File: frontend/src/app/modules/trotamundos/loyality-program/loyality-program.vm.ts
import { StorageVM } from 'app/shared/storage/storage.vm';

export * from 'app/shared/storage/storage.vm';

export class LoyalityProgramVM {
    id: string = null!;
    imageId: string = null!; // Guid referencing StorageVM
    image: StorageVM = null!;
    title: string = null!;
    name: string = null!;
    description: string = null!;

    constructor(init?: Partial<LoyalityProgramVM>) {
        Object.assign(this, init);
    }
}

File: backend/src/trotamundos/lrembecki.core.trotamundos/LoyalityPrograms/LoyalityProgramVM.cs
using lrembecki.core.storage;

namespace lrembecki.core.trotamundos.LoyalityPrograms;

public record LoyalityProgramVM(
    Guid Id,
    Guid ImageId,
    StorageVM Image,
    string Title,
    string Name,
    string Description)
{
    internal static LoyalityProgramVM Map(LoyalityProgramEntity entity)
    {
        if (entity == null)
        {
            return null!;
        }

        return new LoyalityProgramVM(
            entity.Id,
            entity.ImageId,
            StorageVM.Map(entity.Image),
            entity.Title,
            entity.Name,
            entity.Description
        );
    }
}
